plugins/modules/cli_parse.py

```diff
@@ -0,0 +1,275 @@
+        type: str
+        description:
+        - The command to run on the host
+    text:
+        type: str
+        description:
+        - Text to be parsed
+    parser:
+        type: dict
+        description:
+        - Parser specfic parameters
+        suboptions:
+            name:
+                type: str
+                description:
+                - The name of the parser to use
+                required: True
+            command:
+                type: str
+                description:
+                - The command used to locate the parser's template
+            os:
+                type: str
+                description: 
+                - Provide an operating system value to the parser
+            template_path:
+                type: str
+                description:
+                - Path of the parser template on the Ansible controller
+                - This can be a relative or an absolute path
+            vars:
+                type: dict
+                description: 
+                - Additional parser specific parameters
+                - See the cli_parse user guide for examples of parser specific variables
+                - http://link.to.parser.guide 
+ 
+    set_fact:
+        description:
+        - Set the resulting parsed data as a fact
+        type: str
+
+
+notes:
+- The default search path for a parser template is templates/{{ short_os }}_{{ command }}.{{ extension }}
+- => short_os derived from ansible_network_os or ansible_distribution and set to lower case
+- => command is the command passed to the module with spaces replaced with _
+- => extension is specific to the parser used (native=yaml, textfsm=textfsm, ttp=ttp)
+- The default Ansible search path for the templates directory is used for parser templates as well
+- Some parsers may have additional configuration options availble. See the parsers/vars key and the parser's documentation
+- Some parsers require third-party python libraries be installed on the Ansible control node and a specifc python version
+- e.g. Pyats requires pyats and genie and requires Python 3
+- e.g. ntc_templates requires ntc_templates
+- e.g. textfsm requires textfsm
+- e.g. ttp requires ttp
+- e.g. xml requires xml_to_dict
+- Support of 3rd party python libraries is limited to the use of their public APIs as documented
+"""
+
+EXAMPLES = r"""
+
+# Using the native parser
+
+# -------------
+# templates/nxos_show_interface.yaml
+# - example: Ethernet1/1 is up
+#   getval: '(?P<name>\S+) is (?P<oper_state>\S+)'
+#   result:
+#     "{{ name }}":
+#         name: "{{ name }}"
+#         state:
+#         operating: "{{ oper_state }}"
+#   shared: True
+#
+# - example: admin state is up, Dedicated Interface
+#   getval: 'admin state is (?P<admin_state>\S+)'
+#   result:
+#     "{{ name }}":
+#         name: "{{ name }}"
+#         state:
+#         admin: "{{ admin_state }}"
+#
+# - example: "  Hardware: Ethernet, address: 5254.005a.f8b5 (bia 5254.005a.f8b5)"
```

Comment:
```suggestion
# - example: "  Hardware: Ethernet, address: 0000.5E00.5301 (bia 0000.5E00.5301)"
```

Reply:
...assuming those were MAC addresses... I changed to 'documentation safe' mac addresses.